scripts/python-uv.mk

```
@@ -0,0 +1,91 @@
+# UV is required for Python dependency management
+UV ?= uv
+UV_PYTHON = 3.12
+
```

On my local system where I have many pythons managed by uv, when I tried, 
`make -C robot-server dev` I would have issues because even with .python-version file at 3.12 at the root, 3.12 was not being used to spin up the robot server.

I had to solve that by setting it on the make invocation
`make UV_PYTHON=3.12 -C robot-server dev` 

If we export here, recipe commands (including ones that invoke uv without --python $(UV_PYTHON)) like `make -C robot-server dev` will work without having to set it each time.

```suggestion
export UV_PYTHON
```

---

sure, that sounds good. I think we had something similar in the old system with `OT_PYTHON` too, we could keep that around